### DISCUSSION TITLE
How to get global variables that persist after window being closed ?

### DISCUSSION THREAD

**Discussion by andrewberty** [wezterm/wezterm#5563] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### I want to implement a way to write changes to my config files dynamically so it persists (not temporary override).
I was using `sed` to manipulate my files. Now I'm trying to find a way to set global variable that is managed by `wezterm` or `lua` that has some config that I can change on the fly and make it persist after I close wezterm window or reboot my computer. 

Basically, something like `wezterm.GLOBAL` but doesn't clear on reboot. **Any ideas?**

**Comment by wez** [comment]
Sounds like you want a file or a database of somekind.
https://wezfurlong.org/wezterm/config/lua/wezterm.serde/index.html has functions for encoding lua values in various common formats

**Reply by andrewberty** [reply]
Hi Wez, Thanks for your reply,
So if I just created a module called maybe `globals.lua` and returned a table that has my values, I will still need something like `sed` to write to it, right?

**Reply by andrewberty** [reply]
UPDATE:
Thank you so much @wez , I really liked the approach of having a JSON file that has my options to manipulate. I played a little with it and could get it to work with some Input selectors to change colorscheme and font

```lua
M.getLuaFromJSON = function()
	local file = assert(io.open(wezterm.config_dir .. "/globals.json", "r"))
	local json = file:read("a")
	file:close()
	return wezterm.serde.json_decode(json)
end

M.writeLuaToJSON = function(lua)
	local json = wezterm.serde.json_encode_pretty(lua)
	local file = assert(io.open(wezterm.config_dir .. "/globals.json", "w"))
	file:write(json)
	file:close()
end
```

DEMO:

https://github.com/wez/wezterm/assets/95014217/6082bcc6-a103-44f0-819f-39dc71a60056


### At last, I have one more question please,
Is there any chance that I can get input selector to listen for up and down movement events, so I can fire the function on each movement and achieve "live preview"-ish
